google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingHandler.java

```diff
@@ -272,23 +244,55 @@ Formatter getFormatterProperty(String name, Formatter defaultValue) {
    */
   Logging getLogging() {
     if (logging == null) {
-      logging = options.getService();
+      synchronized (this) {
```

Comment:
Well the `close()` method was synchronized already, and the only thing that `close()` protects is the `logging` variable, so it looked like the protection was incomplete to me. Thanks for catching the `volatile` issue, I will fix it.